libraries/AP_HAL_Linux/TCPServerDevice.cpp

```diff
@@ -0,0 +1,138 @@
+#include <AP_HAL.h>
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <netdb.h>
+
+#include "TCPServerDevice.h"
+
+TCPServerDevice::TCPServerDevice(const char *ip, uint16_t port):
+    _ip(ip),
+    _port(port)
+{
+}
+
+TCPServerDevice::~TCPServerDevice()
+{
+
+}
+
+int TCPServerDevice::write(const uint8_t *buf, uint16_t n)
+{
+    return ::send(_net_fd, buf, n, 0);
+}
+
+int TCPServerDevice::read(uint8_t *buf, uint16_t n)
+{
+    return ::recv(_net_fd, buf, n, 0);
+}
+
+bool TCPServerDevice::open()
+{
```

Comment:
I think it would be good to use SocketAPM here, to avoid duplication of socket code